src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -129,33 +267,437 @@ public ResultSet getSchemas() throws SQLException {
 
     @Override
     public ResultSet getCatalogs() throws SQLException {
-        throw new SQLFeatureNotSupportedException("TODO");
+        MongoJsonSchema resultSchema = MongoJsonSchema.createEmptyObjectSchema();
+        resultSchema.addRequiredScalarKeys(new Pair<>(TABLE_CAT, BSON_STRING_TYPE_NAME));
+
+        // All fields in this result set are nested under the bottom namespace.
+        MongoJsonSchema botSchema = MongoJsonSchema.createEmptyObjectSchema();
+        botSchema.properties.put(BOT_NAME, resultSchema);
+
+        BsonExplicitCursor c =
+                new BsonExplicitCursor(
+                        this.conn
+                                .mongoClient
+                                .listDatabaseNames()
+                                .into(new ArrayList<>())
+                                .stream()
+                                .sorted()
+                                .map(
+                                        dbName ->
+                                                createBottomBson(
+                                                        new BsonElement(
+                                                                TABLE_CAT, new BsonString(dbName))))
+                                .collect(Collectors.toList()));
+
+        return new MongoSQLResultSet(null, c, botSchema);
+    }
+
+    /**
```

Comment:
I know javadoc-style comments are necessary for private methods, but figured I'd go all out here to thoroughly explain what this does, why it exists, what its parameters are, what it returns, and when/why it throws a `SQLException`.